tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -832,11 +817,94 @@ public int start() {
+   * {@link YarnConfiguration#RM_PRINCIPAL} property is not set in the
+   * {@code conf} parameter
+   * @throws IOException thrown if there's an error replacing the host name
+   */
+  public static String getRmPrincipal(Configuration conf) throws IOException {
+    String principal = conf.get(YarnConfiguration.RM_PRINCIPAL);
+    String prepared = null;
+
+    if (principal != null) {
+      prepared = getRmPrincipal(principal, conf);
+    }
+
+    return prepared;
+  }
+
+  /**
+   * Perform the <code>_HOST</code> replacement in the {@code principal},
+   * Returning the result. Correctly handles HA resource manager configurations.
+   *
+   * @param rmPrincipal the principal string to prepare
+   * @param conf the configuration
+   * @return the prepared principal string
+   * @throws IOException thrown if there's an error replacing the host name
+   */
+  public static String getRmPrincipal(String rmPrincipal, Configuration conf)
+      throws IOException {
+    if (rmPrincipal == null) {
+      throw new IllegalArgumentException("RM principal string is null");
+    }
+
+    if (HAUtil.isHAEnabled(conf)) {
+      conf = getYarnConfWithRmHaId(conf);
+    }
+
+    String hostname = conf.getSocketAddr(
+        YarnConfiguration.RM_ADDRESS,
+        YarnConfiguration.DEFAULT_RM_ADDRESS,
+        YarnConfiguration.DEFAULT_RM_PORT).getHostName();
+
+    return SecurityUtil.getServerPrincipal(rmPrincipal, hostname);
+  }
+
+  /**
+   * Returns a {@link YarnConfiguration} built from the {@code conf} parameter
+   * that is guaranteed to have the {@link YarnConfiguration#RM_HA_ID}
+   * property set.
+   *
+   * @param conf the base configuration
+   * @return a {@link YarnConfiguration} built from the base
+   * {@link Configuration}
+   * @throws IOException thrown if the {@code conf} parameter contains
+   * inconsistent properties
+   */
+  @VisibleForTesting
+  static YarnConfiguration getYarnConfWithRmHaId(Configuration conf)
+      throws IOException {
+    YarnConfiguration yarnConf = new YarnConfiguration(conf);
+
+    if (yarnConf.get(YarnConfiguration.RM_HA_ID) == null) {
+      // If RM_HA_ID is not configured, use the first of RM_HA_IDS.
+      // Any valid RM HA ID should work.
+      String[] rmIds = yarnConf.getStrings(YarnConfiguration.RM_HA_IDS);
+
+      if ((rmIds != null) && (rmIds.length > 0)) {
+        yarnConf.set(YarnConfiguration.RM_HA_ID, rmIds[0]);
+      } else {
+        throw new IOException("RM_HA_IDS property is not set for HA resource "
+            + "manager");
+      }
+    }
+
+    return yarnConf;
+  }
+
+
   public static void main(String[] args) {
     int exitCode = 0;
 
     // Adds hadoop-inject.xml as a default resource so Azkaban metadata will be present in the new Configuration created
-    HadoopConfigurationInjector.injectResources(new Props() /* ignored */);
+    // HadoopConfigurationInjector.injectResources(new Props() /* ignored */);
```

Comment:
Need to figure out what to do here.

Reply:
What error do you get if you uncomment this line?

Reply:
Was able to uncomment this without any problems.